code/modules/overmap/ships/simulated_ship_data.dm

```diff
@@ -72,7 +72,16 @@
 	if (!isnull(source_template.antag_datum))
 		var/datum/antagonist/ship_datum = new source_template.antag_datum
 		crewmate.mind.add_antag_datum(ship_datum)
-
+/**Checks for verification before being aloud to open a console or object
+  * Arguments:
+  * mob/living/carbon/human/crewmate - The mob being checked for access
+  */
+/obj/structure/overmap/ship/simulated/proc/console_verification(mob/living/carbon/human/crewmate)
+	for (var/datum/weakref/member in crewmembers)
+		if (crewmate == member.resolve())
+			return TRUE
+		else
+			return FALSE
```

Comment:
A couple things, 1. This should be named something more generic like `is_player_in_crew`, because this will be able to be used for a lot of other things too
2. This actually won't work
```suggestion
/obj/structure/overmap/ship/simulated/proc/is_player_in_crew(mob/living/carbon/human/crewmate)
	for (var/datum/weakref/member in crewmembers)
		if (crewmate == member.resolve())
			return TRUE
	return FALSE
```
would work, you don't want to return False in the loop because if the first person in the list isn't who you're looking for it will just assume they're not in the crew. This will go through each member, and if none of them are there it will return false